roles/build-ansible-test-images/tasks/main.yml

```diff
@@ -1,16 +1,30 @@
 - name: Ensure dependencies are available
   include_tasks: pre-requirements.yml
 
-- name: List everything recursively
-  ansible.builtin.command:
-    cmd: "ls -lahR {{ playbook_dir | dirname }}"
-  when: images_build | bool
-
-- name: Build ansible-test container image
-  ansible.builtin.command:
-    cmd: "{{ playbook_dir | dirname }}/{{ item.script }} {{ item.name }}:{{ item.tag }}"
-  when: images_build | bool
-  loop: "{{ images_available }}"
+- when: images_build | bool
+  block:
+    - name: Create temporary directory
+      ansible.builtin.tempfile:
+        state: directory
+        suffix: .test
+      register: remote_tmp_dir
+
+    - name: Copy repository files to temporary directory
+      ansible.builtin.copy:
+        src: '{{ playbook_dir | dirname }}/ansible-test/'
+        dest: '{{ remote_tmp_dir.path }}/ansible-test/'
```

Comment:
Thanks for that.

I'm still confused as to why this is necessary -- I use a VERY similar approach for ara (i.e, https://ansible.softwarefactory-project.io/zuul/build/bc5392b6d7d145ae8a27196f8fdd1068 && https://github.com/ansible-community/ara/blob/master/tests/with_container_images.yaml) and it works but maybe there is something else that copies the repository to the remote nodes first.

Reply:
That was indeed the problem, see #16.

Reply:
OH! :facepalm:

Thanks !